src/reducers/transactionReducer.ts

```diff
@@ -46,6 +46,8 @@ export default function transactionReducer(state: TransactionState, action: Tran
   switch (action.type) {
     case TransactionActionTypes.SET_ESTIMATED_FEE:
       return { ...state, estimatedFee: action.payload.estimatedFee };
+    case TransactionActionTypes.SET_CALCULATING_FEE:
+      return { ...state, isCalculatingFee: action.payload.isCalculatingFee };
```

Comment:
This doesn't make much sense IMHO and that's why I'll keep arguing that reducers in their current form are just boilerplate code (we would achieve exactly the same with `useState` using much less code). 
IMO the whole point of reducer actions is not to expose single-value parameters in a KVDB form, but rather having actions that are specific to your entire data model and ensure that some parameters are updated in unison. Example:
```js
/// Actions
function startEstimateFee() {}
function finishEstimateFee(error: null | string, data: Fee | null) {}

/// Reducers
switch (action.type) {
   case TransactionActionTypes.START_ESTIMATE_FEE:
       return { 
         ...state,
         isCalculatingFee: true,
         estimatedFee: null,
         estimatedFeeError: null
       };
   case TransactionActionTypes.FINISH_ESTIMATE_FEE:
       return {
         ...state,
         isCalculatingFee: false,
         estimatedFee: action.payload.error ? null : action.payload.data,
         estimatedFeeError: action.payload.error,
       };
}
```
This way you can always be sure about some invariants about the set of parameters in your data model. And on the call site you have much less actions to dispatch - basically you separate the logic behind parameters from the place in the code that triggers these updates.

Reply:
> This doesn't make much sense IMHO and that's why I'll keep arguing that reducers in their current form are just boilerplate code (we would achieve exactly the same with useState using much less code).

No. You would have to make sure that every call is wrapped in a useCallback. With dispatchers you don't face that issue.

> IMO the whole point of reducer actions is not to expose single-value parameters in a KVDB form, but rather having actions that are specific to your entire data model and ensure that some parameters are updated in unison.

Very good. Let me see how i can try to switch to this that makes a lot of sense.

Reply:
I like more the name `reset` rather than `start`, because then i can `reset` the values on different part of application if needded.